### DISCUSSION TITLE
How to get a value declared using `const`?

### DISCUSSION THREAD

**Discussion by Tacodiva** [sebastienros/jint#1050] [Q&A] [answered]
Hi, I'm a newbie trying to use Jint and I've encountered a problem. I understand you get values like this:
```C#
Console.WriteLine(new Engine().Execute("var x = 10;").GetValue("x")); // 10
```
But when I try to use the `GetValue` method on a variable declared using `const`, it doesn't seem to work.
```C#
Console.WriteLine(new Engine().Execute("const x = 10;").GetValue("x")); // undefined
```
How do I get the value of `x` when it is a `const`?

**Comment by lahma** [comment, marked as answer]
`GetValue` resolves from object's properties, in this case from global environment's variable environment (which is basically the global object). You need to use `Evaluate` get access items seen by script, in this case the lexical environment which contains let/const/class.

```csharp
Console.WriteLine(new Engine().Execute("const x = 10;").Evaluate("x")); // 10
```
